units/patch_intent.go

```diff
@@ -276,6 +275,17 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		}
 	}
 
+	if j.intent.ReusePreviousPatchDefinition() {
+		previousPatch, err := patch.FindOne(patch.MostRecentPatchByUserAndProject(j.user.Username(), pref.Id))
+		if err != nil {
+			return errors.Wrap(err, "error querying for most recent patch")
+		}
+		if previousPatch == nil {
+			return errors.Errorf("no previous patch available")
+		}
+		patchDoc.VariantsTasks = previousPatch.VariantsTasks
```

Comment:
1) do we need to validate the variantsTasks exist in the new patch's project?
2) if the user specified a particular set of regexes or an alias that matches more tasks in the new patch's project would they want to run those tasks?

Reply:
1. If the patch has been activated it would have to have variants/tasks. 
2. I think it makes sense to just use the same tasks/variants, and if the user wants to add/remove tasks they can. Also as an illustration, let's say for my first patch I used test-alias but then purposefully removed really-long-running-task. The point of --reuse for the second alias, rather than using test-alias again, would be that I don't have to remove really-long-running-task again. This is just supposed to be for convenience, not hacking the system. The definition given for the `reuse` flag is also in line with this.

Reply:
I think it's worth testing in staging to see how this works if you remove some tasks/variants, just so we can document what happens in that scenario. Ideally we would silently not schedule anything that is now invalid

Reply:
That's a good point; I can case on that.

Reply:
The UI did kind of wonky things with missing tasks/variants, so I re-wrote this to check that tasks/variants exist.

First patch: https://evergreen-staging.corp.mongodb.com/version/60271d3297b1d3225462916f

Second patch (coverage removed, one task added to display task and another task removed): https://evergreen-staging.corp.mongodb.com/version/60271dd49dbe323188815c3d